lib/connection.js

```diff
@@ -548,11 +549,13 @@ Connection.prototype.handleRow = function (header, row, meta, rowLength, flags)
     clearTimeout(handler.timeout);
     handler.timeout = null;
   }
-  handler.rowIndex = handler.rowIndex || 0;
-  var rowCallback = handler.options && handler.options.rowCallback;
-  if (rowCallback) {
-    rowCallback(handler.rowIndex++, row, rowLength);
+  if (!handler.rowCallback) {
+    return this.invokeCallback(
+      handler,
+      new errors.DriverInternalError('RowCallback not found for streaming frame handler'));
```

Comment:
Good point, its not possible using the public API, that is the reason it is a `DriverInternalError`.